etl/steps/data/garden/smoking/2024-05-30/cigarette_sales.py

```diff
@@ -0,0 +1,168 @@
+
+    # calculate share of population for each year
+    added_pop = germany_tb[["year", "population"]].groupby("year").sum().reset_index()
+
+    for idx, row in germany_tb.iterrows():
+        germany_tb.loc[idx, "share_of_population"] = (
+            row["population"] / added_pop[added_pop["year"] == row["year"]]["population"].values[0]
+        )
+    # calculate share of cigarettes per adult for weighted average
+    germany_tb[MAN_KEY] = germany_tb[MAN_KEY] * germany_tb["share_of_population"]
+    germany_tb[HAND_KEY] = germany_tb[HAND_KEY] * germany_tb["share_of_population"]
+    germany_tb[TOTAL_KEY] = germany_tb[TOTAL_KEY] * germany_tb["share_of_population"]
+    germany_tb[ALL_KEY] = germany_tb[ALL_KEY] * germany_tb["share_of_population"]
+
+    # sum up values for weighted average
+    germany_tb = germany_tb[COLS_WITH_DATA + ["year"]].groupby("year").sum(min_count=1).reset_index()
+    germany_tb["country"] = "Germany"
+
+    return germany_tb
+
+
+def cast_to_float(df, col):
+    df[col] = df[col].astype("Float64")
+    return df
+
+
+def run(dest_dir: str) -> None:
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("cigarette_sales")
+    # load population data
+    ds_population = paths.load_dataset("population")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["cigarette_sales"].reset_index()
+
+    # Process data.
+
+    # Fix years column (change dtype to integer and expand timeframes)
+    df_years_fixed = Table(standardise_years(tb), metadata=tb.metadata)
+    # replace table with dataframe with fixed years, concat with empty df to keep metadata
+    tb = pr.concat([tb[0:0], df_years_fixed])
+
+    # convert million to actual values
+    tb["manufactured_cigarettes"] = tb["manufactured_cigarettes_millions"] * 1000000
+    tb["handrolled_cigarettes"] = tb["handrolled_cigarettes_millions"] * 1000000
+    tb["total_cigarettes"] = tb["total_cigarettes_millions"] * 1000000
+
+    tb = tb.drop(
+        columns=[
+            "manufactured_cigarettes_millions",
+            "handrolled_cigarettes_millions",
+            "total_cigarettes_millions",
+        ]
+    )
+
+    # Calculate weighted average for Germany 1950-1990
+    germany_tb = include_split_germany(tb, ds_population)
+
+    # include data for Germany 1950-1990
+    tb = pr.concat([tb, germany_tb])
+
+    # reorder columns
+    tb = tb[["year", "country"] + COLS_WITH_DATA]
+
+    # drop rows with no data (nan values)
+    tb = tb.dropna(how="all", subset=COLS_WITH_DATA)
+
+    # drop rows with zeros and NaNs
+    tb_temp = tb.fillna(0)
+    rows_to_drop = tb_temp[(tb_temp[COLS_WITH_DATA] == 0).all(axis=1)].index
+    tb = tb.drop(rows_to_drop)
+
+    # cast columns to float
+    for col in COLS_WITH_DATA:
+        tb = cast_to_float(tb, col)
+
+    # harmonize countries (also exclude east and west germany)
+    tb = geo.harmonize_countries(
```

Comment:
Yeah, that makes sense! I excluded them from the table after adding the combined column - I was try to sort of "preserve" them in case somebody is interested specifically in data from East/ West Germany but I agree that the trade-off there is probably not worth it.